From e5c87fd02a62dc540d97200f612d40811efe07c3 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Wed, 23 Jan 2019 16:28:36 +0200
Subject: [PATCH 036/235] sdk: dts: ls104x: drop smmu from the qds and usdpaa
 sdk dts

Also drop the smmu from the qds and usdpaa versions of the
SDK device trees because SMMU is supported only for the
upstream version of the dpaa ethernet drivers.

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1043a-qds-sdk.dts     | 14 ++++++++++++++
 .../boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts  | 14 ++++++++++++++
 .../boot/dts/freescale/fsl-ls1046a-qds-sdk.dts     | 14 ++++++++++++++
 .../boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts  | 14 ++++++++++++++
 4 files changed, 56 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
index 493a7942db01..5dddc5e6ec4c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-qds-sdk.dts
@@ -64,6 +64,20 @@
 &soc {
 #include "qoriq-dpaa-eth.dtsi"
 #include "qoriq-fman3-0-6oh.dtsi"
+
+pcie@3400000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3500000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3600000 {
+       /delete-property/ iommu-map;
+};
+
+/delete-node/ iommu@9000000;
 };
 
 &fman0 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
index 5940dc680227..f6e714d0d4f7 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a-rdb-usdpaa.dts
@@ -92,6 +92,20 @@
 			fsl,fman-oh-port = <&fman0_oh2>;
 		};
 	};
+
+	pcie@3400000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3500000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3600000 {
+	       /delete-property/ iommu-map;
+	};
+
+	/delete-node/ iommu@9000000;
 };
 / {
 	reserved-memory {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
index bb0f062bc5ac..1bbf6bb58a8d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-qds-sdk.dts
@@ -64,6 +64,20 @@
 &soc {
 #include "qoriq-dpaa-eth.dtsi"
 #include "qoriq-fman3-0-6oh.dtsi"
+
+pcie@3400000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3500000 {
+       /delete-property/ iommu-map;
+};
+
+pcie@3600000 {
+       /delete-property/ iommu-map;
+};
+
+/delete-node/ iommu@9000000;
 };
 
 &fsldpaa {
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
index 953b20d2e76f..540289c3ab11 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1046a-rdb-usdpaa.dts
@@ -85,6 +85,20 @@
 			fsl,fman-oh-port = <&fman0_oh2>;
 		};
 	};
+
+	pcie@3400000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3500000 {
+	       /delete-property/ iommu-map;
+	};
+
+	pcie@3600000 {
+	       /delete-property/ iommu-map;
+	};
+
+	/delete-node/ iommu@9000000;
 };
 / {
 	reserved-memory {
-- 
2.17.1

